src_c/math.c

```diff
@@ -1633,18 +1633,49 @@ vector_reflect_ip(pgVector *self, PyObject *normal)
 static double
 _vector_distance_helper(pgVector *self, PyObject *other)
 {
-    Py_ssize_t i;
-    double distance_squared, tmp;
+    Py_ssize_t i, dim = self->dim;
+    double distance_squared = 0;
+
+    if (pgVector_Check(other)) {
+        pgVector *otherv = (pgVector *)other;
+        double dx, dy;
+
+        if (dim != otherv->dim) {
+            PyErr_SetString(PyExc_TypeError, "Vectors must be the same size");
+            return -1;
+        }
 
-    distance_squared = 0;
-    for (i = 0; i < self->dim; ++i) {
-        tmp = PySequence_GetItem_AsDouble(other, i) - self->coords[i];
-        distance_squared += tmp * tmp;
+        dx = otherv->coords[0] - self->coords[0];
+        dy = otherv->coords[1] - self->coords[1];
+
+        distance_squared = dx * dx + dy * dy;
+
+        if (dim == 3) {
+            double dz;
+            dz = otherv->coords[2] - self->coords[2];
+            distance_squared += dz * dz;
+        }
     }
-    /* PySequence_GetItem_AsDouble can fail in which case it will set an Err */
-    if (PyErr_Occurred())
-        return -1;
+    else {
+        double tmp;
+        if (!(PyTuple_Check(other) || PyList_Check(other))) {
+            PyErr_SetString(PyExc_TypeError, "A sequence was expected");
+            return -1;
+        }
```

Comment:
```suggestion
        PyObject *fast_seq = PySequence_Fast(other, "A sequence was expected");
        if (!fast_seq) {
            return -1;
        }
```

I think we should be supporting generic sequences here, not just tuples and lists... I wouldn't mind if doing so reduces performance for this case by a small bit.
The code I suggested should work for all sequences (not sure about other things like generators and non-sequence iterables, but the old code did not support em anyways) and I'm pretty sure this also has an internal cpython fastpath for tuples and lists

Reply:
I said it once and i'll say it again. PySequence_Fast also converts strings into "PySequence_Fast objects", and implementing what you asked me( i did it and tested it) only gives you a segfault. At that point to resolve it you have to add a string check and exit with error if that's the case.
But at that point you are doing the same as before (basically a two checks + one conversion) so you are back where you started. I don't agree with this and just like you said:

> not sure about other things like generators and non-sequence iterables, but the old code did not support em anyways

yes you are right here that we did not support them before so the current way is clearly the best one, just 2 fast checks amd everything stays the same as before.

Reply:
actually i think i finally understood how this works xD

Reply:
Strings are sequences, `PySequence_Fast` is doing the right thing. But ofc, `PyFloat_AsDouble` should correctly set an error if it gets characters.

Reply:
yeah now it works fine, just needed to use the converted string sequence